android/core/src/main/java/com/stadiamaps/ferrostar/core/FerrostarCore.kt

```diff
@@ -299,9 +301,11 @@ class FerrostarCore(
     }
   }
 
-  fun stopNavigation() {
+  fun stopNavigation(stopLocationUpdates: Boolean = true) {
     foregroundServiceManager?.stopService()
-    locationProvider.removeListener(this)
+    if (stopLocationUpdates) {
+      locationProvider.removeListener(this)
```

Comment:
New argument that doesn't break source compatibility; lets us keep the location provider running, which makes life easier for single map view apps.